### PR TITLE
build: generate api docs for all tertiary entry-points

### PR DESCRIPTION
**Pull request by devversion**
Updates our Dgeni setup to generate API docs for all entry-points
which are configured in the `cdk/config.bzl` or `material/config.bzl` file. 

This basically means that we now generate API docs for `material/*/testing`
entry-points and for `cdk/testing/*`.

Removes the code that made `cdk/testing` a separate sub package. This
involves more code than just treating it as "real" entry-point (for which we generate bundles etc.).

### CODE REVIEW

File: tools/dgeni/common/class-inheritance.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import {ClassLikeExportDoc} from 'dgeni-packages/typescript/api-doc-types/ClassLikeExportDoc';
+
+/** Gets all class like export documents which the given doc inherits from. */
+export function  getInheritedDocsOfClass(doc: ClassLikeExportDoc): ClassLikeExportDoc[] {
```
Comment (josephperrott) [inline tools/dgeni/common/class-inheritance.ts]: extra space between `function` and `getInheritedDocsOfClass`

File: src/cdk/config.bzl [hunk 2]
```diff
@@ -18,13 +18,12 @@ CDK_ENTRYPOINTS = [
...
+    "testing/testbed",
 ]
 
-# List of all entry-point targets of the Angular Material package. Note that
-# we do not want to include "testing" here as it will be treated as a standalone
-# sub-package of the "ng_package".
-CDK_TARGETS = ["//src/cdk"] + \
-              ["//src/cdk/%s" % ep for ep in CDK_ENTRYPOINTS if not ep == "testing"]
+# List of all entry-point targets of the Angular Material package.
+CDK_TARGETS = ["//src/cdk"] + ["//src/cdk/%s" % ep for ep in CDK_ENTRYPOINTS]
```
Comment (mmalerba) [inline src/cdk/config.bzl]: It looks like this used to exclude `testing` but now does not. This is intentional right?

File: src/cdk/testing/BUILD.bazel [hunk 3]
```diff
@@ -15,15 +14,6 @@ ts_library(
     module_name = "@angular/cdk/testing",
 )
 
-npm_package(
```
Comment (mmalerba) [inline src/cdk/testing/BUILD.bazel]: Why was this removed? I thought it was necessary for cdk/testing to be included in the release?

File: src/cdk/testing/protractor/BUILD.bazel [hunk 4]
```diff
@@ -14,3 +14,8 @@ ts_library(
         "@npm//protractor",
     ],
 )
+
+filegroup(
+    name = "source-files",
+    srcs = glob(["**/*.ts"]),
```
Comment (mmalerba) [inline src/cdk/testing/protractor/BUILD.bazel]: For all of these, should we exclude `**/*.spec.ts`?

File: src/cdk/config.bzl (hunk 2 above)
Comment (devversion) [reply src/cdk/config.bzl]: Yes, I added a message for it in the PR description. I made `cdk/testing/*` real entry-points. Previously it was treated as sub-`ng_package`, but that rather made this config more confusing and less consistent. So now it's more consistent I'd say 😄

File: src/cdk/testing/BUILD.bazel (hunk 3 above)
Comment (devversion) [reply src/cdk/testing/BUILD.bazel]: See comment above. We are including `cdk/testing` and `cdk/testing/*` like all other entry-points now. i.e. doing bundling, downleveling etc.

It made configuration easier and also I don't see a benefit in making it inconsistent with what we do for `material/*/testing` entry-points.

File: src/cdk/testing/protractor/BUILD.bazel (hunk 4 above)
Comment (devversion) [reply src/cdk/testing/protractor/BUILD.bazel]: Yes. technically you are right. Those shouldn't be needed for docs generation, but it will make the +100 `source-files` filegroups more verbose. so maybe fix that up in the future?

Comment (devversion) [reply src/cdk/testing/protractor/BUILD.bazel]: Somehow it would feel a bit weird adding the exclusion glob here though? I mean, then we have another target in the build file that needs to know about spec files. We should figure something out, but probably not related to this PR.

Comment (devversion) [reply src/cdk/testing/protractor/BUILD.bazel]: As discussed on Slack. This is not an issue now, nor does it have any big negative effect. Though we should consider cleaning this up in the future: COMP-238
